inlang/source-code/sdk/src/persistence/slotfiles/createSlotWriter.test.ts

```diff
@@ -0,0 +1,145 @@
+import { expect, describe, it } from "vitest"
+import createSlotWriter from "./createSlotWriter.js"
+import { createNodeishMemoryFs } from "@lix-js/fs"
+
+type DocumentExample = {
+	id: string
+	content: string
+}
+
+// TODO write test if new file gets created on collison
+// TODO test that inserting the same item fails on memory
+// TODO test that inserting the same item (id) fails between processes
+// TODO test that data witten by another process reaches the process via fs events
+
+describe("Disconnected slot storage", () => {
+	it("insert should lead to a transient record", async () => {
+		const memoryFs = createNodeishMemoryFs()
+
+		const slotStorage = await createSlotWriter<DocumentExample>({
+			fs: memoryFs,
+			path: "/slot/",
+			fileNameCharacters: 3,
+			slotsPerFile: 16,
+			watch: false,
+		})
+		const insertedDocument: DocumentExample = {
+			id: "1",
+			content: "first document",
+		}
+		await slotStorage.insert(insertedDocument, false)
+
+		expect(slotStorage._writerInternals.transientSlotEntries.size).eq(1)
+		const docs1 = await slotStorage.findDocumentsById([insertedDocument.id])
```

Comment:
what else would you find here? getAll() or findById should be more readable and precise enough